### PR TITLE
Gate async HTTP on a config

### PR DESCRIPTION
**Pull request by jonathanmetzman**
The async HTTP usage has been shown to cause OOMs in OSS-Fuzz that cause significant problems,
in particular causing NTP to fail to sync.

### CODE REVIEW

File: src/clusterfuzz/_internal/google_cloud_utils/storage.py [hunk 1]
```diff
@@ -1153,6 +1158,17 @@ def blobs_bucket():
   return local_config.ProjectConfig().get('blobs.bucket')
 
 
+def use_async_http():
+  global _USES_ASYNC_HTTP
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/google_cloud_utils/storage.py:1162]: nit: just use `memoize` here. 

e.g. https://github.com/google/clusterfuzz/blob/2d57351ad8e22597a94849db6e013b12d9dcdc3c/src/clusterfuzz/_internal/base/utils.py#L904

**Review by oliverchang — APPROVED** [review]

File: src/clusterfuzz/_internal/google_cloud_utils/storage.py (hunk 1 above)
Comment (jonathanmetzman) [reply src/clusterfuzz/_internal/google_cloud_utils/storage.py:1162]: I couldn't figure out if it works with no arguments and it seems a little less readable to be honest. I'll switch it if you think it works.

Comment (oliverchang) [reply src/clusterfuzz/_internal/google_cloud_utils/storage.py:1162]: it should work -- the example linked has no arguments.

**Review by vitorguidi — APPROVED** [review]
lgtm

**Pull request merged**